sql/backup_from_standby.sh

```diff
@@ -203,9 +222,31 @@ ${BACKUP_RESTORE_COMMAND}
 ${BACKUP_TARGET_XID_COMMAND}
 EOF
 pg_ctl start -w -t 600 -D ${PGDATA_PATH} > /dev/null 2>&1
-psql -p ${TEST_PGPORT} --no-psqlrc -d pgbench -c "SELECT * FROM pgbench_branches;" > ${TEST_BASE}/TEST-0002-after.out
+get_database_data > ${TEST_BASE}/TEST-0002-after.out
 diff ${TEST_BASE}/TEST-0002-before.out ${TEST_BASE}/TEST-0002-after.out
+echo ''
 
+# check to start as primary (stand-alone) if the recovery target is latest
```

Comment:
Wouldn't it be better to perform a restore on the standby side for this regresstion test?
I think the goal of this regression test is to restore the database cluster where standby.signal exists, the restore from the primary side does not seem to be related to your opinion.

Reply:
Thanks for your comments! 

Sorry, I couldn't understand yet. Although I agree with your comments, I thought the test is to perform a restore using standby-side data?

1. start primary and standby
2. backup from standby side
3. stop primary
4. restore the data directory using the backup data got in the 2nd phase
5. start using the data directory

Do you mean that data directory must be standby server's one in the 4th and the 5th phase? For example, is it better to change ${PGDATA_PATH} to ${SBYDATA_PATH}?
> pg_rman restore -B ${BACKUP_PATH} -D ${PGDATA_PATH} --quiet;echo $?
> sed -i -e "s/^port = .*$/port = ${TEST_PGPORT}/" ${PGDATA_PATH}/postgresql.conf
> pg_ctl start -w -t 600 -D ${PGDATA_PATH} > /dev/null 2>&1

If I'm missing something, please let me know.

Reply:
Oh... I'm sorry.
I imagined that using the backup file acquired from the standby side would be restored on the primary and standby sides, thought that a regression test was necessary.
In other words, my imagined "pg_ctl START -d SBYDATA_PATH" after copying the restored database cluster to "SBYDATA_PATH."

However, the pg_rman manual wrote.
"Basically, the backup from standby-site is used for restoring on MASTER. pg_rman doesn't treat the backup as restoring on standby automatically." 
In other words, it means "basically does not support restore from standby", so I don't think you need to add this test.

In other words, please ignore the comments on the regression test that I have requested.
Please apologize for making such a strange comment.